bin/dartdoc.dart

```diff
@@ -252,26 +248,30 @@ main(List<String> arguments) async {
     ]);
   }
 
-  setConfig(
-      addCrossdart: args['add-crossdart'],
-      examplePathPrefix: args['example-path-prefix'],
-      showWarnings: args['show-warnings'],
-      includeSource: args['include-source'],
-      inputDir: inputDir,
-      sdkVersion: sdk.sdkVersion,
-      autoIncludeDependencies: args['auto-include-dependencies'],
-      packageOrder: args['package-order'].isEmpty
-          ? args['category-order']
-          : args['package-order'],
-      reexportMinConfidence:
-          double.parse(args['ambiguous-reexport-scorer-min-confidence']),
-      verboseWarnings: args['verbose-warnings'],
-      excludePackages: args['exclude-packages'],
-      dropTextFrom: dropTextFrom,
-      validateLinks: args['validate-links']);
-
-  DartDoc dartdoc = new DartDoc(inputDir, excludeLibraries, sdkDir, generators,
-      outputDir, packageMeta, includeLibraries, includeExternals);
+  DartDocConfig config = new DartDocConfig.fromParameters(
```

Comment:
@jcollins-g – FYI https://pub.dartlang.org/packages/build_cli 😄

Reply:
@kevmoo  Hmmm. What I'm really looking for is a non-throwing, tolerant version of safe_config thrown in a blender with build_cli and a sprinkle of the analyzer's directory-specific configuration semantics.  I have that a few PRs out on this line of development.